src/compiler/parser.ts

```diff
@@ -5055,13 +5055,23 @@ namespace ts {
         // STATEMENTS
         function parseBlock(ignoreMissingOpenBrace: boolean, diagnosticMessage?: DiagnosticMessage): Block {
             const node = <Block>createNode(SyntaxKind.Block);
+            const openBracePosition = scanner.getTokenPos();
             if (parseExpected(SyntaxKind.OpenBraceToken, diagnosticMessage) || ignoreMissingOpenBrace) {
                 if (scanner.hasPrecedingLineBreak()) {
                     node.multiLine = true;
                 }
 
                 node.statements = parseList(ParsingContext.BlockStatements, parseStatement);
-                parseExpected(SyntaxKind.CloseBraceToken);
+                if (token() !== SyntaxKind.CloseBraceToken) {
+                    const lastError = lastOrUndefined(parseDiagnostics);
+                    const start = scanner.getTokenPos();
+                    if (!lastError || start !== lastError.start) {
+                        parseErrorAt(openBracePosition, openBracePosition + 1, Diagnostics.The_parser_expected_to_find_a_to_match_the_token_here);
+                    }
+                }
+                else {
+                    parseExpected(SyntaxKind.CloseBraceToken)
+                }

```

Comment:
Rather than this, since we want a `related` span (not a change to the original span), consider:
```ts
if (!parseExpected(SyntaxKind.CloseBraceToken)) {
    const lastError = lastOrUndefined(parseDiagnostics)!;
    addRelatedInfo(lastError, createFileDiagnostic(sourceFile, openBracePosition, 1, Diagnostics.The_parser_expected_to_find_a_to_match_the_token_here));
}
```

Reply:
Thanks for the quick response. I've introduced the requested changes and fixed the linting errors.